### PR TITLE
Improvement to servatrice database interface

### PR DESCRIPTION
**Pull request by ctrlaltca**
The main change here is the reuse of prepared statements. Servatrice already uses them, but at every query it creates a new prepared statement and discards it at the end of the query. After this change, the prepared statements are cached and reused. This should hopefully lower the load on the database.
Plus, there are two minor fixes:
 * due to an unnecessary exec() call, the "sessions" and "users" tables were always locked/unlocked twice.
 * a query used the hardcoded "cockatrice_" table prefix instead of the correct value specified in the config.

Even if i tested these changes myself, i'll be glad if someone could test this on an actual server with more than 2 users.

### CODE REVIEW

**Comment by Daenyth** [comment]
@woogerboy21 can we try this on the backup server?

File: servatrice/src/servatrice.cpp [hunk 1]
```diff
@@ -447,14 +443,13 @@ void Servatrice::statusUpdate()
...
 
-    QSqlQuery query(servatriceDatabaseInterface->getDatabase());
-    query.prepare("insert into " + dbPrefix + "_uptime (id_server, timest, uptime, users_count, games_count, tx_bytes, rx_bytes) values(:id, NOW(), :uptime, :users_count, :games_count, :tx, :rx)");
-    query.bindValue(":id", serverId);
-    query.bindValue(":uptime", uptime);
-    query.bindValue(":users_count", uc);
-    query.bindValue(":games_count", gc);
-    query.bindValue(":tx", tx);
-    query.bindValue(":rx", rx);
+    QSqlQuery *query = servatriceDatabaseInterface->prepareQuery("insert into {prefix}_uptime (id_server, timest, uptime, users_count, games_count, tx_bytes, rx_bytes) values(:id, NOW(), :uptime, :users_count, :games_count, :tx, :rx)");
```
Comment (Daenyth) [inline servatrice/src/servatrice.cpp]: In the future we can probably make this configurable to reduce load

**Comment by woogerboy21** [comment]
Sure I can throw it on a little later.  Is there any type of user load that we would like to see it put against?

**Comment by ctrlaltca** [comment]
Nice, thank you!

**Comment by woogerboy21** [comment]
Anyone have any idea on how one would go about measuring the performance impact of the change?

**Comment by Daenyth** [comment]
Io should improve or be on par with before. CPU should decrease or be
similar. Nothing should perform worse

**Comment by woogerboy21** [comment]
Alright, this is now on the backup server as well.  As far as measuring performance, I was hoping there was more of some type of mysql type query / indicator we could use as in quantity of table locks or some statistical value to compare a before and after.  I guess if it works and the underlying code is better its a step in the right direction.

**Comment by Daenyth** [comment]
I'm not familiar enough to say

**Comment by woogerboy21** [comment]
Im seeing this on some of the logins:
```
Thu Feb 12 12:42:34 2015 Server::loginUser: ServerSocketInterface(0x9b58c0) name= "zioteromporculo"
Thu Feb 12 12:42:34 2015 session id: 638553
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range
Thu Feb 12 12:42:34 2015 QMYSQLResult::data: column 7 out of range

```

**Comment by ctrlaltca** [comment]
Its seems like the problem is due to 69e6aa355ff9caa016f066447b2d3aad5978dbb8 that added in Servatrice_DatabaseInterface::evalUserQueryResult() the parsing of the 7th column of the query result.
Servatrice_DatabaseInterface::evalUserQueryResult() is a generic parsing method that is used in three different places in the code (getUserData / getBuddyList / getIgnoreList), but only one of the three corresponding queries has been updated to add the 7th column.
So a solution could be: 
 * adding a dummy 7th column to the other 2 queries, or
 * split the method to two different methods, or
 * add a "bool parse7thColumn" parameter to the method;

Any preference?

**Comment by Daenyth** [comment]
I'd prefer two methods

**Comment by ctrlaltca** [comment]
After checking the involved code a bit better, i found that a "bool complete" parameter already existed with the very same purpose. So, i reordered the columns in the queries and used that parameter to skip parsing the unused column.

**Comment by woogerboy21** [comment]
So prior to the reordering, other than the thrown errors would there be any other negative impacts you can think of that may occur?

**Comment by ctrlaltca** [comment]
No negative impact our of the error message in the log: that column was not used in the buddy/ignored lists.

**Comment by ctrlaltca** [comment]
afaik this is running since at least a week on both the main and the backup server. @woogerboy21 did you notice any problem or change in server load?

**Comment by Daenyth** [comment]
If this has been running on the main server I think it's ready to merge

**Comment by woogerboy21** [comment]
Correct, its been running with out a problem (beyond the first column 7 error that was corrected).  And I have not seen any issue's so far.

**Pull request merged**